website/docs/basics/blocks-intro.md

```diff
@@ -195,4 +195,32 @@ And folder structure `src/Blocks/blocks` is kinda weird.
 
 You can, but you shouldn't. We use this key to provide the CSS selectors on all our custom blocks in the block editor. Additional styles are added to this selector for the wrapper to work on full width and remove some of the native block editor styles. 
 
-So don't remove or change this 😅 .
+So don't remove or change this 😅.
+
+### Simple vs Compound blocks
+
+In a nutshell, there is no difference between blocks. The WordPress core block is a block, but we like to make a distinction between what is simple and what is a compound block.
+
+**Simple block** is a block that is used just as is and provides a small functionality like heading, paragraph, button, etc.
+
+**Compound block** is a block that builds from multiple components and provides a bigger functionality.
+
+### Naming is hard
+
+*There are only two hard things in Computer Science: cache invalidation and naming things. - Phil Karlton*
+
+Yes, naming is hard, and no matter how long is your development experience you will always struggle with names for your components, block, files, variables, functions, etc.
+
+Here are some of our recommendations to ease your pain:
+* You should never name your component based on its usage. For example, you have a component card for custom post-type books, you should never call this component `card-book` bat rather use a more generic name like `card-product`.
+* Make your names as generic as possible for better reusability.
+* Always think about the future. How can this feature be used in some other way?
+* When naming your attributes ask yourself: "is this attribute going to be used in any other way?". For example, you have an attribute for adding font-weight: bold to your text. You can create an attribute called font-weight and set it as a `boolean` type and that will be ok if you have only one font-weight. A better way would be to put it as a string and provide and SelectControl component if there is any possibility that in the future you will have additional font-weight.
+* Name booleans positively. Example: `isValid`, `isLoading`, `isComplete`.
+* Don’t hesitate to use longer names: We use short words even if it doesn’t provide full insight about the code which is a bad thing to do.
```

Comment:
> We use short words even if it doesn’t provide full insight about the code which is a bad thing to do.

Do we need to open an issue for this? :D

Reply:
Yeah, kinda goes against what you are saying in this chapter 😅

Reply:
fixed